src/Compilers/CSharp/Portable/BoundTree/BoundExpression.cs

```diff
@@ -167,6 +167,34 @@ public override Symbol ExpressionSymbol
         // DevDiv 1087283 tracks deciding whether or not to refactor this into BoundNodes.xml.
         public ImmutableArray<PropertySymbol> OriginalIndexersOpt { get; private set; }
 
+        public BoundIndexerAccess Update(bool useSetterForDefaultArgumentGeneration)
+        {
+            if (useSetterForDefaultArgumentGeneration != this.UseSetterForDefaultArgumentGeneration)
```

Comment:
Nit: consider flipping the logic around to bail out fast (if equal, then return this).

Reply:
> Nit: consider flipping the logic around to bail out fast (if equal, then return this).

Simply follow the pattern for the way other Update methods are implemented.